lib/Elastica/Query.php

```diff
@@ -64,21 +64,16 @@ public static function create($query)
             case $query instanceof self:
                 return $query;
             case $query instanceof AbstractQuery:
-                return new self($query);
-            case empty($query):
-                return new self(new MatchAll());
             case \is_array($query):
-                return new self($query);
-            case \is_string($query):
-                return new self(new QueryString($query));
-            case $query instanceof AbstractSuggest:
-                return new self(new Suggest($query));
-
             case $query instanceof Suggest:
-                return new self($query);
-
             case $query instanceof Collapse:
-                return new self($query);
+                return new static($query);
+            case empty($query):
```

Comment:
Is there a reason you reordered these statements?

Reply:
to avoid code duplication. but you're right, this changes semantics as will. i will revert that change.